common/network_manager.go

```diff
@@ -73,6 +74,12 @@ func NewNetworkManager(
 	ctx context.Context, session *Session, manager *FrameManager, parent *NetworkManager,
 ) (*NetworkManager, error) {
 	state := k6lib.GetState(ctx)
+
+	resolver, err := newResolver(state.Options.DNS)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
`newResolver` doesn't resolve anything, just creates a new `Resolver` instance. And an error is returned only if `parseTTL()` fails, so as long as that error is clear, this can remain returning `err`.

Reply:
It's better to follow the error chain on an error, that's why I suggested it. What about?

```suggestion
	resolver, err := newResolver(state.Options.DNS)
	if err != nil {
		return nil, fmt.Errof("newResolver(%+v): %w", state.Options.DNS, err)
	}
```

Reply:
This will be extra noisy with the `parseTTL` change, especially with `%+v` of `Options.DNS`, but sure, I changed it in 24b1fce.

Reply:
Same: https://github.com/grafana/xk6-browser/pull/205#discussion_r797333757